src/main/java/games/strategy/engine/lobby/server/userDB/DbUserController.java

```diff
@@ -18,39 +19,57 @@
  *  This is a migration tool, to go from one data source to another. When we switch over completely
  *  we can simplify and write/read to the primary datasource directly.
  */
+// TODO: rename to: UserController
 public class DbUserController implements UserDao {
   private static final Logger logger = Logger.getLogger(DbUserController.class.getName());
 
-  private static final Collection<UserDaoPrimarySecondary> implementations = Arrays.asList(
-      new DerbyUserController(),
-      new SqlUserController()
-  );
-
   /**
    * Stores a convenience reference to the authoritative DAO implementation.
    * We use this datasource for reading, we always write to it, any errors during read/write are considered
    * critical (on the other hand secondary DAO read/write errors can be logged and ignored)
    */
-  private static final UserDao primaryDao = determinePrimaryDataSource();
+  private final UserDao primaryDao;
+  private final Collection<UserDao> secondaryDaoSet;
+
+
+  /**
+   * The default constructor uses a production configuration.
+   */
+  public DbUserController() {
+    this(
+        new DerbyUserController(
+            UserDaoPrimarySecondary.Role.PRIMARY,
+            () -> Database.getDerbyConnection()),
+        new DerbyUserController(
+            UserDaoPrimarySecondary.Role.SECONDARY,
+            () -> Database.getPostgresConnection()));
+  }
 
-  private static UserDao determinePrimaryDataSource() {
-    final List<UserDao> primaryDataSourceCandidates = implementations.stream()
+  /**
+   * Finds the primary dao implementation and verifies there is only one.
+   * Everything else is stored as secondary, we will parallel write to secondary (ignoring errors),
+   * but will only read from the primary.
+   */
+  @VisibleForTesting DbUserController(UserDaoPrimarySecondary ... userDaos) {
+    final List<UserDao> primaryDaoList = Arrays.stream(userDaos)
         .filter(UserDaoPrimarySecondary::isPrimary)
         .collect(Collectors.toList());
-
     Preconditions.checkState(
-        primaryDataSourceCandidates.size() == 1,
+        primaryDaoList.size() == 1,
         "Startup Assumption - exactly one UserDao implementation is 'primary'");
+    primaryDao = primaryDaoList.get(0);
 
-    return primaryDataSourceCandidates.get(0);
+    secondaryDaoSet = Arrays.stream(userDaos)
+        .filter(dao -> !dao.isPrimary())
```

Comment:
Nitpicking... to be symmetric with line 55, this line can be replaced with `.filter(not(UserDaoPrimarySecondary::isPrimary))` where `not` is imported as

```java
import static games.strategy.util.PredicateUtils.not;
```